File: DataStructure/20. Valid Parentheses.py
from collections import deque
class Solution:
    def isValid(self, s: str) -> bool:
        stack = deque()
        brakets={'(' : ')', '{' : '}', '[' : ']'}

        for braket in s:
            if braket in brakets:
                stack.append(brakets[braket])
            else:
                if stack and braket == stack[-1]:
                    stack.pop()
                else:
                    return False
        
        if stack:
            return False
        
        return True

File: LeetCode/DataStructure/20. Valid Parentheses.cs
public class Solution {
    public bool IsValid(string s) {
        Stack<char> stack = new Stack<char>();
        Dictionary<char, char> brakets = new Dictionary<char, char>
        { 
            {'(', ')'},
            {'{', '}'}, 
            {'[', ']'}
        };
        
        foreach (char c in s)
        {
            if (brakets.ContainsKey(c))
            {
                stack.Push(brakets[c]);
            }
            else
            {
                if (stack.Count > 0 && c == stack.Peek())
                {
                    stack.Pop();
                }
                else 
                {
                    return false;
                }
            }
        }
        
        if (stack.Count > 0)
        {
            return false;    
        }
        return true;
    }
}